src/Domain/Block.cpp

```diff
@@ -141,23 +173,38 @@ bool operator==(const Block<VolumeDim>& lhs, const Block<VolumeDim>& rhs) {
   // Since the external boundary conditions are all abstract base classes we
   // can't compare them, but we check that the typeid matches between LHS and
   // RHS.
-  return lhs.id() == rhs.id() and lhs.neighbors() == rhs.neighbors() and
-         lhs.external_boundaries() == rhs.external_boundaries() and
-         alg::all_of(
-             rhs.external_boundaries(),
-             [&lhs, &rhs](const Direction<VolumeDim>& dir) {
-               return lhs.external_boundary_conditions().contains(dir) and
-                      rhs.external_boundary_conditions().contains(dir) and
-                      typeid(lhs.external_boundary_conditions().at(dir)) ==
-                          typeid(rhs.external_boundary_conditions().at(dir));
-             }) and
-         lhs.is_time_dependent() == rhs.is_time_dependent() and
-         (lhs.is_time_dependent()
-              ? (lhs.moving_mesh_logical_to_grid_map() ==
-                     rhs.moving_mesh_logical_to_grid_map() and
-                 lhs.moving_mesh_grid_to_inertial_map() ==
-                     rhs.moving_mesh_grid_to_inertial_map())
-              : lhs.stationary_map() == rhs.stationary_map());
+  bool blocks_are_equal =
+      (lhs.id() == rhs.id() and lhs.neighbors() == rhs.neighbors() and
+       lhs.external_boundaries() == rhs.external_boundaries() and
+       alg::all_of(
+           rhs.external_boundaries(),
+           [&lhs, &rhs](const Direction<VolumeDim>& dir) {
+             return lhs.external_boundary_conditions().contains(dir) and
+                    rhs.external_boundary_conditions().contains(dir) and
+                    typeid(lhs.external_boundary_conditions().at(dir)) ==
+                        typeid(rhs.external_boundary_conditions().at(dir));
+           }) and
+       lhs.is_time_dependent() == rhs.is_time_dependent());
+
+  if (lhs.is_time_dependent() and not lhs.has_distorted_frame()) {
+    blocks_are_equal =
+        blocks_are_equal and (lhs.moving_mesh_logical_to_grid_map() ==
+                                  rhs.moving_mesh_logical_to_grid_map() and
+                              lhs.moving_mesh_grid_to_inertial_map() ==
+                                  rhs.moving_mesh_grid_to_inertial_map());
+  } else if (lhs.is_time_dependent() and lhs.has_distorted_frame()) {
+    blocks_are_equal = blocks_are_equal and
+                       (lhs.moving_mesh_logical_to_grid_map() ==
+                        rhs.moving_mesh_logical_to_grid_map()) and
+                       (lhs.moving_mesh_grid_to_distorted_map() ==
+                            rhs.moving_mesh_grid_to_distorted_map() and
+                        lhs.moving_mesh_distorted_to_inertial_map() ==
+                            rhs.moving_mesh_distorted_to_inertial_map());
```

Comment:
(Question) If there is a distorted frame, Is it possible to have `grid_to_distorted` and `distorted_to_inertial` maps along with `grid_to_inertial` maps? Or if there is a distorted frame, do all maps need to go to this intermediary distorted frame?

Reply:
Yes, we should be able to define a `grid_to_inertial`, which will be the same as `grid_to_distorted` composed with `distorted_to_inertial`.  So `grid_to_inertial` could be a useful shortcut.  Having `grid_to_inertial` (even when a distorted frame exists) would simplify the parts of the code that don't care about the distorted frame; i.e. everything except for the horizon finders and the control systems.